kimchi/src/circuits/polynomials/foreign_field_add/gadget.rs

```diff
@@ -399,34 +406,3 @@ fn set_up_lookup_env_data<F: PrimeField>(
         joint_lookup_table_d8,
     })
 }
-
-/// Get array of foreign field addition circuit gate types
-pub fn circuit_gates() -> [GateType; GATE_COUNT] {
```

Comment:
I think you should probably leave these interfaces and also use them from proof systems.  If we have all gates use them, then that means all gates/gadgets have the same interface.  This means that the proof system code is always be the same, which means that perhaps one day we can make the proof system hook up to all the gates in a generic way, rather than each differently.  I was sort of moving things in this direction by abstracting things in this way and then using them from proof systems.

Reply:
I understand, but I am removing after Matthew's suggestion to delete this for single-gate gadgets

Reply:
Ah, ok - he knows better! 👍🏼